cloudinit/distros/__init__.py

```diff
@@ -259,6 +259,9 @@ def _apply_hostname(self, hostname):
     def _select_hostname(self, hostname, fqdn):
         # Prefer the short hostname over the long
         # fully qualified domain name
+        if util.get_cfg_option_bool(self._cfg, "prefer_fqdn_over_hostname",
```

Comment:
Thanks for the PR @hamalq, this functionality needs unittest change
You can find some example tests in [tests/unittests/test_handler/test_handler_set_hostname.py](https://github.com/canonical/cloud-init/blob/master/tests/unittests/test_handler/test_handler_set_hostname.py) that might give you a hint on how to exercise this aspect of cloud-init. In might be nice to have this unittest coverage added to cloudinit/distros/tests/test_init.py which covers some of the functions in cloudinit.distros.__init__ module anyway.